File: 14.最长公共前缀.js
/*
 * @lc app=leetcode.cn id=14 lang=javascript
 *
 * [14] 最长公共前缀
 */

// @lc code=start
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  if (strs === null || strs.length === 0) {
    return "";
  }
  let pre = strs[0];
  for (let i = 1; i < strs.length; i++) {
    while (!strs[i].startsWith(pre)) {
      pre = pre.substring(0, pre.length - 1);
    }
  }
  return pre;
};
// @lc code=end

File: 14.最长公共前缀.rs
/*
 * @lc app=leetcode.cn id=14 lang=rust
 *
 * [14] 最长公共前缀
 */

// @lc code=start
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.len() == 0 || &strs[0] == "" {
            return String::new();
        }
        let mut n: usize = 1;
        loop {
            for word in strs.iter() {
                if n > word.len() || &strs[0][0..n] != &word[0..n] {
                    return String::from(&strs[0][0..n - 1]);
                }
            }
            n += 1;
        }
    }
}
// @lc code=end

